types/leaflet/index.d.ts

```diff
@@ -583,7 +584,7 @@ export interface PolylineOptions extends PathOptions {
 export class Polyline<T extends geojson.GeometryObject = geojson.LineString | geojson.MultiLineString, P = any> extends Path {
     constructor(latlngs: LatLngExpression[], options?: PolylineOptions);
     toGeoJSON(): geojson.Feature<T, P>;
-    getLatLngs(): LatLng[];
+    getLatLngs(): LatLng[] | LatLng[][] | LatLng[][][];
```

Comment:
According to the [Leaflet documentation](http://leafletjs.com/reference-1.3.0.html#polyline-getlatlngs) it should return `LatLng[]` for simple polylines and `LatLng[][]` for multi-polylines. But you are also allowed to nest them as you with. For example also this type would be processed `LatLng[][][][][]`, but I think it makes no sense to make a long list of possible nesting. So in my opinions the return type should be: `LatLng[] | LatLng[][]`! It covers both different kinds of polylines and multi-polylines.

@mcauer Do you have  a reason for `LatLng[][][]`?

Another approach would be a type like mentioned below, but as far as I know a circular type is not allowed:

```typescript
type NestedPolyline = LatLng[] | NestedPolyline[];
```

Reply:
I agree for Polylines. But as the method is also inherited by Polygons. A multipolygon with holes may return LatLng[][][]. See examples of Multipolygon: http://leafletjs.com/reference-1.3.0.html#polygon
Or can we have different return types for the getLatLngs() method for Polylines and Polygons?